tests/console/docker.pm

```diff
@@ -50,47 +50,62 @@ sub run {
     my $alpine_image_version = '3.5';
     assert_script_run("docker pull alpine:$alpine_image_version", 300);
-    assert_script_run("docker images -q alpine:$alpine_image_version");
-    #   - pull hello-world image, typical docker demo image
+    assert_script_run("docker images alpine:$alpine_image_version | grep alpine");
+    #   - pull typical docker demo image without tag. Should be latest.
     #   - https://store.docker.com/images/hello-world
     assert_script_run("docker pull hello-world", 300);
-    assert_script_run("docker images -q hello-world:latest");
+    assert_script_run("docker images hello-world | grep latest");
 
     # local images can be listed
-    validate_script_output("docker images", sub { m/alpine/ });
+    my $cmd_docker_images = 'docker images';
+    my $listed_images     = script_output($cmd_docker_images);
+    unless ($listed_images =~ m/alpine/ && $listed_images =~ m/hello-world/) {
+        die("error: missing image: $cmd_docker_images");
+    }
+
 
     # containers can be spawned
-    validate_script_output("docker run --name test_1 hello-world",                                                     sub { m/Hello from Docker/ });
-    validate_script_output("docker run --name test_2 alpine:$alpine_image_version /bin/echo Hello world",              sub { m/Hello world/ });
-    validate_script_output("docker ps -a",                                                                             sub { m/test_1/ });
-    validate_script_output("docker ps -a",                                                                             sub { m/test_2/ });
-    validate_script_output("docker run --name test_ephemeral --rm alpine:$alpine_image_version /bin/echo Hello world", sub { m/Hello world/ });
+    assert_script_run('docker run --name test_1 hello-world | grep "Hello from Docker\!"');
+    assert_script_run(qq#docker run --name test_2 alpine:$alpine_image_version /bin/echo Hello world | grep "Hello world"#);
+    assert_script_run(qq#docker run --name test_ephemeral --rm alpine:$alpine_image_version /bin/echo Hello world | grep "Hello world"#);
+    my $cmd_docker_ps_a   = 'docker ps -a';
+    my $output_containers = script_output($cmd_docker_ps_a);
+    unless ($output_containers =~ m/test_1/ && $output_containers =~ m/test_2/) {
+        die("error: missing container: $cmd_docker_ps_a");
+    }
+    die("error: ephemeral container was not removed: $cmd_docker_ps_a") if ($output_containers =~ m/test_ephemeral/);
 
     # containers can be run on background
-    my ($container_id) = script_output("docker run -d -t -i alpine:$alpine_image_version /bin/sh") =~ /(.+)/;
-    script_run("docker inspect --format='{{.State.Running}}' ${container_id}");
-    validate_script_output("docker inspect --format='{{.State.Running}}' ${container_id}", sub { m/true/ });
+    my ($container_id) = script_output("docker run -d alpine:$alpine_image_version tail -f /dev/null") =~ /(.+)/;
```

Comment:
Since you are touching this part of the code, please use the new docker commands:

`docker container run` instead of `docker run`